pandas/core/indexes/base.py

```diff
@@ -800,24 +806,21 @@ def __deepcopy__(self, memo=None):
             memo = {}
         return self.copy(deep=True)
 
-    def _validate_names(self, name=None, names=None, deep=False):
+    def _validate_names(self, name=None, names=None):
         """
-        Handles the quirks of having a singular 'name' parameter for general
-        Index and plural 'names' parameter for MultiIndex.
+        Returns valid `names` considering both `names` and `name`
+        or raises TypeError in case of wrong arguments passed.
         """
-        from copy import deepcopy
         if names is not None and name is not None:
-            raise TypeError("Can only provide one of `names` and `name`")
-        elif names is None and name is None:
-            return deepcopy(self.names) if deep else self.names
-        elif names is not None:
-            if not is_list_like(names):
-                raise TypeError("Must pass list-like as `names`.")
-            return names
-        else:
-            if not is_list_like(name):
-                return [name]
-            return name
+            raise TypeError("Can provide only one of names and name arguments")
+
+        if names is not None and not is_list_like(names):
+            raise TypeError("names must be list-like")
+
+        if name is not None:
```

Comment:
you might be able to remove this as ``.set_names`` already handles the list-like / scalar case (or maybe remove it there)